submissions/2024-12-12-OpenFF-Sage-2.0.0-Training-Optimization-Dataset-v1.0/generate-combined-dataset.py

```diff
@@ -0,0 +1,149 @@
+
+import numpy as np
+import requests
+import logging
+logging.getLogger("openff").setLevel(logging.ERROR)
+
+from openff.qcsubmit.results import OptimizationResultCollection
+from openff.qcsubmit.datasets import OptimizationDataset
+from openff.qcsubmit.factories import OptimizationDatasetFactory
+from openff.qcsubmit.common_structures import MoleculeAttributes
+
+from openff.toolkit.utils import RDKitToolkitWrapper, ToolkitRegistry
+from openff.units import unit
+
+file = requests.get("https://raw.githubusercontent.com/openforcefield/openff-sage/refs/heads/main/data-set-curation/quantum-chemical/data-sets/1-2-0-opt-set-v3.json")
+filtered_and_combined = OptimizationResultCollection.parse_raw(file.content)
+rec_and_mol = filtered_and_combined.to_records()
+
+print('Number of results: ',filtered_and_combined.n_results,flush=True)
+print('Finished converting to records',flush = True)
+
+# Need to input initial molecules so that QCA will recognize the inputs as the same.
+initial_mols = [rec[0].initial_molecule for rec in rec_and_mol]
+
+dataset_factory1 = OptimizationDatasetFactory()
+provenance1 = dataset_factory1.provenance(ToolkitRegistry([RDKitToolkitWrapper]))
```

Comment:
Just looking at your environment, it looks like OpenEye is installed. Do you have a license? If so, it takes precedence over RDKit so probably best to add that here too.

Reply:
Just following up on this, it's less important if CMILES isn't being generated, but if RDKit is there OpenEye should be considered too.